cellarium/ml/core/module.py

```diff
@@ -62,6 +71,8 @@ def __init__(
             warnings.filterwarnings("ignore", message="Attribute 'model' is an instance of `nn.Module`")
             self.save_hyperparameters(logger=False)
         self.pipeline: CellariumPipeline | None = None
+        self._lightning_training_using_datamodule: bool = False
+        self._cpu_transforms_in_module_pipeline: bool = not self._lightning_training_using_datamodule
```

Comment:
Can we leave only one boolean flag here? Having two flags is unnecessary and can potentially lead to a bug. Then `remove_cpu_transforms_from_module_pipeline` can be removed as well because  setting `self._lightning_training_using_datamodule = True` essentially does it. Or if you want to remove `self._lightning_training_using_datamodule` and keep `remove_cpu_transforms_from_module_pipeline` then make it a private method because it should only be called by the `configure_model` and not by the user directly.

Reply:
We can, I just am not sure what to call the flag.  I wanted to call it `_lightning_training_using_datamodule`, but then if I let the flag be set by callable methods, it's no longer guaranteed to be what the name implies.

Maybe I'll remove those helper methods and just act directly on `_lightning_training_using_datamodule`.

Reply:
Either name is fine, `_cpu_transforms_in_module_pipeline` sounds slightly better to me :)

> Maybe I'll remove those helper methods

Yeah, I agree. The flag is changed only in one place so having an entire method to do it might be too much.

Reply:
Okay I'll change it to `_cpu_transforms_in_module_pipeline` which is more helpful in terms of "what the flag does".  `_lightning_training_using_datamodule` is more helpful in terms of what sets this flag, but eh, maybe less readable and less useful.